### ISSUE TITLE
I3D crashes at the last average pooling 3D if the network has a shape < 2 in its temporal dimension

### ISSUE THREAD

**Issue by mourad1081** [deepmind/kinetics-i3d#75]
At this line:
`net = tf.nn.avg_pool3d(net, ksize=[1, 2, 7, 7, 1], strides=[1, 1, 1, 1, 1], padding=snt.VALID)`

I3D suddenly crashes. I was extracting RGB features, then at some point, my script crashed at my 98th video. I ran it again, and it crashes at that line and for all the following videos too.

I have checked the shape of the variable `net` just before doing that instruction and I have noticed that it had a shape of (1, 1, 7, 7, 1024), which is < 2 in the temporal component. 

**I added this to fix the issue but I don't know if I am correct by doing so:**

```
@Edit: oddly, 1, 2, 7, 7, 1 does not work..., what if I try avg on t=1 ?
if net.shape[1] < 2:
    net = tf.nn.avg_pool3d(net, ksize=[1, 1, 7, 7, 1], strides=[1, 1, 1, 1, 1], padding=snt.VALID)
else:
    net = tf.nn.avg_pool3d(net, ksize=[1, 2, 7, 7, 1], strides=[1, 1, 1, 1, 1], padding=snt.VALID)
```

Is it okay? I don't know why I3D crashes since I do not edited its architecture before that line and it worked well on previous videos.

All the video are .mp4, 320x240, resized to ndarray of 1xTx224x224x3.

**Comment by joaoluiscarreira** [comment]
You could replace padding=snt.VALID to padding=snt.SAME, then if number of
frames is smaller than 2, the video is padded.

Joao

**Comment by mourad1081** [comment]
Thank you, I'll try that.
